aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -1113,6 +1130,115 @@ std::tuple<Tensor&,Tensor&> qr_out(Tensor& Q, Tensor& R, const Tensor& self, boo
   return at::linalg_qr_out(Q, R, self, mode);
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ orgqr ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+DEFINE_DISPATCH(orgqr_stub);
+
+/*
+  The orgqr function allows reconstruction of an orthogonal (or unitary) matrix Q,
+  from a sequence of elementary reflectors, such as is produced by the geqrf function.
+
+  Args:
+  * `input` - Tensor with the directions of the elementary reflectors below the diagonal.
+  * `tau` - Tensor containing the magnitudes of the elementary reflectors.
+  * `result` - result Tensor, which will contain the orthogonal (or unitary) matrix Q.
+  * `infos` - Tensor to store LAPACK/MAGMA error codes
+
+  For further details, please see the LAPACK/MAGMA documentation.
+*/
+Tensor& orgqr_out_info(const Tensor& input, const Tensor& tau, Tensor& result, Tensor& infos) {
+  TORCH_INTERNAL_ASSERT(input.dim() >= 2);
+  TORCH_INTERNAL_ASSERT(input.size(-2) >= input.size(-1));
+  TORCH_INTERNAL_ASSERT(input.size(-1) >= tau.size(-1));
+
+  TORCH_INTERNAL_ASSERT(input.scalar_type() == tau.scalar_type());
+  TORCH_INTERNAL_ASSERT(input.device() == tau.device());
+
+  TORCH_INTERNAL_ASSERT(result.scalar_type() == input.scalar_type());
+  TORCH_INTERNAL_ASSERT(result.device() == input.device());
+
+  TORCH_INTERNAL_ASSERT(infos.scalar_type() == at::kInt);
+  TORCH_INTERNAL_ASSERT(infos.numel() == std::max<int64_t>(1, batchCount(input)));
+
+  // if result has no elements we can modify it
+  if (result.numel() == 0) {
+    at::native::resize_as_(result, input.transpose(-2, -1), MemoryFormat::Contiguous);
+    result.transpose_(-2, -1);
+  } else {
+    // Resize messes up the strides and we expect strictly column major order, so let's not use at::native::resize_output
```

Comment:
We should use resize_output for now instead of hard erroring in this case.

Reply:
Yes, this comment shouldn't be here now. I removed it and moved the assert out of the "else" branch.